tools/make_unstruct_grid/MakeUnstructGrid.py

```diff
@@ -0,0 +1,315 @@
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+import matplotlib
+import matplotlib.dates as mdates
+import sys
+import code  # For development: code.interact(local=locals())  code.interact(local=dict(globals(), **locals()))
+import argparse
+import math
+from scipy.io import netcdf as nc
+import xml.etree.ElementTree as et
+
+# The user specifies a couplet of domain/surface files
+# from which they want to base their new unstructured grid. Then they provide
+# a list of geographic coordinates in latitude and longitude. These coordinates
+# are sampled from the base dataset, and written to an output dataset.
+#
+# This method is certainly useful for generating small sets of unstructured
+# grid-cells. It may not be the best method for generating large sets. One
+# may want to use "ncks" (ie Charlie Zender's NCO tools) for subsetting large
+# grids. This method will arrange the new grid cells in a 1D vector, and assumes
+# the input grids are based on a 2d array of cells.
+#
+# This method will assume that the grid-cell extents of the new unstructured grids
+# match the extents of the base files. If you want finer or coarser resolution,
+# just dig up a different base file.
+#
+# This method uses nearest neighbor.
+#
+# This may have trouble on newer surface files, particularly if they have topo
+# unit information.  It won't be difficult to add that type of functionality
+# if it doesn't work, I (Ryan) just haven't tried it.
+#
+# All controls over this process can be found in the xml control file. See
+# andes7x7.xml for an example.
+
+# Usage MakeUnstructGrid.py --fin=xmlfile.xml
+
+def TransferData(da_key,ds_base,ds_unst,minis,minjs,dset_type):
+        
+    print('  Transferring: {}'.format(da_key))
+
+    if(dset_type=='domain'):
+        xname = 'nj'
+        yname = 'ni'
+        ny = len(minis)  #nj = len(minis)
+        nx = 1           #ni
+        nv = 4
+    elif(dset_type=='surface'):
+        xname = 'lsmlat'
+        yname = 'lsmlon'
+        ny = len(minis)
+        nx = 1
+        
+        #numurbl = 3 ;
+        #nlevurb = 5 ;
+        #numrad = 2 ;
+        #nchar = 256 ;
+        #nlevsoi = 10 ;
+        #time = UNLIMITED ; // (12 currently)
+        #lsmpft = 17 ;
+        #natpft = 17 ;
+
+
+    
+    # Determin the data type
+    if(ds_base[da_key].dtype == 'float64'):
+        dtype_out = np.float64
+    elif(ds_base[da_key].dtype == 'int32'):
+        dtype_out = np.int32
+    else:
+        print('unknown data type: {}.\n Exiting'.format(ds_base[da_key].dtype))
```

Comment:
See line 72 comment below:
```suggestion
    elif(ds_base[da_key].dtype == 'float32'):
        dtype_out = np.float32
    else:
        print('unknown data type: {}.\n Exiting'.format(ds_base[da_key].dtype))
```